# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/CSSEditUtils.cpp
# Commit: 876b1031a6ae
# Full Hash: 876b1031a6ae41a87ed146976f279b96bde9c7f6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-03 21:40:10
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 1: Make each transaction class grab their members with local variable before touching the DOM tree r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69302
# ==============================================================================

diff -r 7f069649d6cf -r 876b1031a6ae editor/libeditor/CSSEditUtils.cpp
--- a/editor/libeditor/CSSEditUtils.cpp	Fri Apr 03 06:03:25 2020 +0300
+++ b/editor/libeditor/CSSEditUtils.cpp	Fri Apr 03 01:38:22 2020 +0000
@@ -383,8 +383,12 @@
 nsresult CSSEditUtils::SetCSSProperty(Element& aElement, nsAtom& aProperty,
                                       const nsAString& aValue,
                                       bool aSuppressTxn) {
+  nsCOMPtr<nsStyledElement> styledElement = do_QueryInterface(&aElement);
+  if (NS_WARN_IF(!styledElement)) {
+    return NS_ERROR_INVALID_ARG;
+  }
   RefPtr<ChangeStyleTransaction> transaction =
-      ChangeStyleTransaction::Create(aElement, aProperty, aValue);
+      ChangeStyleTransaction::Create(*styledElement, aProperty, aValue);
   if (aSuppressTxn) {
     return transaction->DoTransaction();
   }
@@ -416,8 +420,12 @@
 nsresult CSSEditUtils::RemoveCSSProperty(Element& aElement, nsAtom& aProperty,
                                          const nsAString& aValue,
                                          bool aSuppressTxn) {
+  nsCOMPtr<nsStyledElement> styledElement = do_QueryInterface(&aElement);
+  if (NS_WARN_IF(!styledElement)) {
+    return NS_ERROR_INVALID_ARG;
+  }
   RefPtr<ChangeStyleTransaction> transaction =
-      ChangeStyleTransaction::CreateToRemove(aElement, aProperty, aValue);
+      ChangeStyleTransaction::CreateToRemove(*styledElement, aProperty, aValue);
   if (aSuppressTxn) {
     nsresult rv = transaction->DoTransaction();
     NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),